simulator/Fetcher.ts

```diff
@@ -10,16 +10,23 @@ export class Fetcher implements IFetcher {
         var wi = state.warriorIndex;
         var warrior = state.warriors[wi];
 
+        // skip dead warriors
+        if(warrior.tasks.length === 0) {
+            wi = (wi + 1) % state.warriors.length
```

Comment:
Will this crap out if there are two consecutive dead warriors?

Reply:
Ah yes, good point. I did wonder about that, perhaps loop into a checkdead function would be better